libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -852,9 +859,13 @@ void AP_DDS_Client::on_request(uxrSession* uxr_session, uxrObjectId object_id, u
         if (deserialize_success == false) {
             break;
         }
-        mode_switch_response.status = AP::vehicle()->set_mode(mode_switch_request.mode, ModeReason::DDS_COMMAND);
         mode_switch_response.curr_mode = AP::vehicle()->get_mode();
-
+        if (external_control->is_enabled()) {
```

Comment:
Is this really the best way to go about enabling and disabling external control?  It's going to be a lot of these calls - flash and easy to mess up.

Did you consider putting this on the deserialize part instead?

Reply:
Are you saying we reject any input data in AP_DDS if the aux function is triggered? If we did that, it would also affect things like AP_DDS visual odom, or state requests that don't change behavior of the autopilot such as `pre_arm_check`. 

I was thinking about this yesterday, and it seems there's nothing the autopilot can do now to protect against bad visual odom data either other than the user switch to a mode that doesn't use it. 

I'm open to alternatives that still satisfy the user story. The calls need to go somewhere, whether it's in AP_DDS, or here. Last dev call, the team seemed to be on board with the approach, specifically for blocking control commands.

Reply:
Well, I'm not going to block this as others have agreed to it.

But I'd suggest a whitelist of `object_id.id` which you check if control is disabled, checked before hitting the `switch` statement.  Probably done as an additional `switch` statement in a separate method, `use_object_id_when_control_disabled` or something.

Reply:
Ah yea, that seems like a good alternative. @tizianofiorenzani Can you try a separate implementation of that and compare the flash? It would be good to know which approach is better before we get too deep in adding all the rest of the interfaces.